File: Frontend/src/models/UserProfile.ts
import { User } from "./User";

export enum Gender {
    Female,
    Male,
    Other,
}

export enum MaritalStatus {
    Single,
    Married,
    Widowed,
    Separated,
    Divorced,
}

export interface UserProfile {
    id?: number;

    userId?: number;
    user?: User;

    bio: string;
    location: string;

    birthday?: string;
    gender: Gender;
    maritalStatus: MaritalStatus;

    pagePreference?: number;
}

File: Backend/MuseumAPI/Models/UserProfile.cs
﻿namespace MuseumAPI.Models
{
    public enum MaritalStatus
    {
        Single,
        Married,
        Widowed,
        Separated,
        Divorced
    }

    public enum Gender
    {
        Female,
        Male,
        Other
    }

    public class UserProfile
    {
        public virtual long Id { get; set; }

        public virtual long? UserId { get; set; }
        public virtual User User { get; set; } = null!;
        public virtual string? Bio { get; set; }
        public virtual string? Location { get; set; }
        public virtual DateTime? Birthday { get; set; }
        public virtual Gender Gender { get; set; }
        public virtual MaritalStatus MaritalStatus { get; set; }

        public virtual long PagePreference { get; set; }
    }
}
